courier/message.go

```diff
@@ -16,14 +21,146 @@ const (
 	MessageStatusAbandoned
 )
 
+const (
+	messageStatusQueuedText     = "queued"
+	messageStatusSentText       = "sent"
+	messageStatusProcessingText = "processing"
+	messageStatusAbandonedText  = "abandoned"
+)
+
+func ToMessageStatus(str string) (MessageStatus, error) {
+	switch str {
+	case messageStatusQueuedText:
+		return MessageStatusQueued, nil
+	case messageStatusSentText:
+		return MessageStatusSent, nil
+	case messageStatusProcessingText:
+		return MessageStatusProcessing, nil
+	case messageStatusAbandonedText:
+		return MessageStatusAbandoned, nil
+	default:
+		return MessageStatus(0), errors.New("Message status is not valid")
+	}
```

Comment:
```suggestion
	switch s := stringsx.SwitchExact(str); {
	case s.AddCase(MessageStatusQueued.String()):
		return MessageStatusQueued, nil
	case s.AddCase(MessageStatusSent.String()):
		return MessageStatusSent, nil
	case s.AddCase(MessageStatusProcessing.String()):
		return MessageStatusProcessing, nil
	case s.AddCase(MessageStatusAbandoned.String()):
		return MessageStatusAbandoned, nil
	default:
		return 0, errors.WithStack(herodot.ErrBadRequest.WithError(s.ToUnknownCaseErr()).WithReason("Message status is not valid"))
	}
```

Reply:
Using `herodot` errors automatically sets the status code correctly.

Reply:
@zepatrik I guess `ErrBadRequest.WithError` should be `ErrBadRequest.WithWrap` right ?